### ISSUE TITLE
concatenate small vector to image at every pixel

### ISSUE THREAD

**Issue by ahundt** [fchollet/keras#7706]
I have an image of unknown dimension `(None, None, None, 3)` and a small 1d vector of size 5, and I'd like to concatenate this vector to the channels of every pixel in the image, thus creating an "image" of size `(None, None, None, 8)`. 

How can I do this concatenation, and is it best to do with tensors or should I do it with layers? 

I've been looking at `K.tile` but the documentation is a bit weak for that function.

In other words, I’ve got a 1d tensor of length w I would like to concatenate to every pixel in an image, which would make an nxmx3 image become nxmx(3+w) image.

**Comment by kgrm** [comment]
```
im = K.variable(np.random.rand(32, 128, 128, 3))
v = K.variable(np.random.rand(5))
v_rep = K.repeat_elements(v[None, None, None, :], im.shape[0], 0)
v_rep = K.repeat_elements(v_rep, im.shape[1], 1)
v_rep = K.repeat_elements(v_rep, im.shape[2], 2)
print v_rep.eval().shape # (32, 128, 128, 5)
concatenated = K.concatenate([im, v_rep], axis=3)
print concatenated.eval().shape # 32, 128, 128, 8
```

**Comment by ahundt** [comment]
I ended up just fixing the dimensions for the time being. K.tile works pretty well, it just repeats the entire contents of the data in the axis dimensions you indicate.
